app/src/main/java/com/github/se/eduverse/ui/archive/ArchiveScreen.kt

```diff
@@ -1,72 +1,164 @@
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Scaffold
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.runtime.collectAsState
 import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.unit.dp
 import com.github.se.eduverse.isNetworkAvailable
 import com.github.se.eduverse.model.Folder
+import com.github.se.eduverse.ui.DeleteFoldersDialog
 import com.github.se.eduverse.ui.navigation.BottomNavigationMenu
 import com.github.se.eduverse.ui.navigation.LIST_TOP_LEVEL_DESTINATION
 import com.github.se.eduverse.ui.navigation.NavigationActions
 import com.github.se.eduverse.ui.navigation.Route
 import com.github.se.eduverse.ui.navigation.TopNavigationBar
 import com.github.se.eduverse.viewmodel.FolderViewModel
 
+@OptIn(ExperimentalFoundationApi::class)
 @Composable
 fun ArchiveScreen(
     navigationActions: NavigationActions,
     folderViewModel: FolderViewModel,
     context: Context = LocalContext.current
 ) {
   val folders: List<Folder> by folderViewModel.folders.collectAsState()
+  var isSelectMode by remember { mutableStateOf(false) }
+  var selected by remember { mutableStateOf(emptyList<Int>()) }
+  var deleteDialogOpen by remember { mutableStateOf(false) }
 
   LaunchedEffect(Unit) { folderViewModel.getArchivedUserFolders() }
 
   Scaffold(
-      topBar = { TopNavigationBar("Archived Courses", navigationActions) },
+      topBar = {
+        TopNavigationBar("Archived Courses", navigationActions) {
+          if (isSelectMode) {
+            IconButton(
+                onClick = {
+                  if (context.isNetworkAvailable()) {
+                    deleteDialogOpen = true
+                  } else {
+                    folderViewModel.showOfflineMessage(context)
+                  }
+                },
+                modifier = Modifier.testTag("delete")) {
+                  Icon(imageVector = Icons.Default.Delete, contentDescription = "Delete")
+                }
+          }
+        }
+      },
       bottomBar = {
         BottomNavigationMenu({ navigationActions.navigateTo(it) }, LIST_TOP_LEVEL_DESTINATION, "")
       }) { padding ->
+        if (deleteDialogOpen) {
+          DeleteFoldersDialog(
+              number = selected.size,
+              onDismiss = { deleteDialogOpen = false },
+              onConfirm = {
+                deleteDialogOpen = false
+                isSelectMode = false
+                selected.forEach { folderViewModel.deleteFolder(folders[it]) }
+                selected = emptyList()
+              })
+        }
```

Comment:
selected.forEach { folderViewModel.deleteFolder(folders[it]) } performs a folder-by-folder deletion. This could cause performance issues or partial errors if a deletion fails.